lib/ohai/plugins/linux/platform.rb

```diff
@@ -74,12 +86,16 @@ def os_release_file_is_cisco?
       platform get_redhatish_platform(contents)
       platform_version contents.match(/(\d\.\d\.\d)/)[0]
     elsif File.exist?("/etc/redhat-release")
-      if File.exist?('/etc/os-release') && (os_release_info = os_release_file_is_cisco? ) # check if Cisco
-        platform os_release_info['ID']
-        platform_family os_release_info['ID_LIKE']
-        platform_version os_release_info['VERSION'] || ""
+      contents = File.read("/etc/redhat-release").chomp
+      if (os_release_info = os_release_file_is_cisco? ) # check if Cisco
```

Comment:
Seems reasonable. We were trying to work within the existing paradigm of the code but I agree that your proposed change may be more sensible.